userns.go

```diff
@@ -0,0 +1,220 @@
+package storage
+
+import (
+	"os"
+	"os/user"
+	"strconv"
+
+	"github.com/containers/common/pkg/unshare"
+	"github.com/containers/storage/pkg/idtools"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func getAutoUserNsMappings(username string) ([]idtools.IDMap, []idtools.IDMap, error) {
+	var uids, gids []idtools.IDMap
+
+	if unshare.IsRootless() {
+		username = os.Getenv("USER")
+		if username == "" {
+			var id string
+			if os.Geteuid() == 0 {
+				id = strconv.Itoa(unshare.GetRootlessUID())
+			} else {
+				id = strconv.Itoa(os.Geteuid())
+			}
+			userID, err := user.LookupId(id)
+			if err == nil {
+				username = userID.Username
+			}
+		}
+	} else {
+		if username == "" {
+			username = "storage"
```

Comment:
Containers, and should be a constant.

Reply:
it is the name we are already using for `RemapUser`.

I am fine with changing it but I think we should be consistent.  Should we change the default for `RemapUser` too?

Reply:
SGTM